// reakit/es/Popover/index.js
import 'reakit-system/createComponent';
import 'reakit-system/createHook';
import '../Box/Box.js';
import '../_rollupPluginBabelHelpers-419c3ce6.js';
import 'react';
import 'reakit-utils/useForkRef';
import 'reakit-utils/isButton';
import 'reakit-warning';
import 'reakit-utils/useLiveRef';
import 'reakit-utils/isSelfTarget';
import 'reakit-utils/useIsomorphicEffect';
import 'reakit-utils/hasFocusWithin';
import 'reakit-utils/isPortalEvent';
import '../Tabbable/Tabbable.js';
import '../Clickable/Clickable.js';
import '../Button/Button.js';
import 'reakit-utils/useSealedState';
import 'reakit-system/useCreateElement';
import 'reakit-utils/getDocument';
import '../Id/IdProvider.js';
import '../Id/IdState.js';
import 'reakit-utils/getActiveElement';
import 'reakit-utils/tabbable';
import '../Disclosure/DisclosureState.js';
import '../Disclosure/DisclosureContent.js';
import 'react-dom';
import '../Portal/Portal.js';
import 'reakit-utils/removeItemFromArray';
import '../MenuContext-eb1aa1c9.js';
import '../Dialog/Dialog.js';
import 'body-scroll-lock';
import 'reakit-utils/useUpdateEffect';
import 'reakit-utils/closest';
import 'reakit-utils/contains';
import '../Dialog/DialogState.js';
import '../DialogBackdropContext-2db2c622.js';
import '../Dialog/DialogBackdrop.js';
import '../Disclosure/Disclosure.js';
import '../Dialog/DialogDisclosure.js';
import '@popperjs/core';
import 'reakit-utils/shallowEqual';
export { usePopoverState } from './PopoverState.js';
export { Popover, usePopover } from './Popover.js';
export { PopoverArrow, usePopoverArrow } from './PopoverArrow.js';
export { PopoverDisclosure, usePopoverDisclosure } from './PopoverDisclosure.js';
export { PopoverBackdrop, usePopoverBackdrop } from './PopoverBackdrop.js';
