// reakit/es/Tooltip/index.js
import 'reakit-system/createComponent';
import 'reakit-system/createHook';
import '../Box/Box.js';
import '../_rollupPluginBabelHelpers-419c3ce6.js';
import 'react';
import 'reakit-utils/useForkRef';
import 'reakit-utils/useLiveRef';
import 'reakit-utils/isSelfTarget';
import 'reakit-utils/useIsomorphicEffect';
import 'reakit-utils/useSealedState';
import 'reakit-utils/getDocument';
import '../Id/IdProvider.js';
import '../Id/IdState.js';
import '../Disclosure/DisclosureState.js';
import '../Disclosure/DisclosureContent.js';
import 'react-dom';
import '../Portal/Portal.js';
import '../Dialog/DialogState.js';
import '@popperjs/core';
import 'reakit-utils/shallowEqual';
import '../Popover/PopoverState.js';
import '../Popover/PopoverArrow.js';
export { u as useTooltipState } from '../TooltipState-17ceb977.js';
export { Tooltip, useTooltip } from './Tooltip.js';
export { TooltipArrow, useTooltipArrow } from './TooltipArrow.js';
export { TooltipReference, useTooltipReference } from './TooltipReference.js';
