// reakit/es/Tooltip/TooltipArrow.js
import { createComponent } from 'reakit-system/createComponent';
import { createHook } from 'reakit-system/createHook';
import '../Box/Box.js';
import { _ as _objectWithoutPropertiesLoose, a as _objectSpread2 } from '../_rollupPluginBabelHelpers-419c3ce6.js';
import 'react';
import 'reakit-utils/useForkRef';
import 'reakit-utils/useIsomorphicEffect';
import 'reakit-utils/useSealedState';
import '../Id/IdProvider.js';
import '../Id/IdState.js';
import '../Disclosure/DisclosureState.js';
import '../Dialog/DialogState.js';
import '@popperjs/core';
import 'reakit-utils/shallowEqual';
import '../Popover/PopoverState.js';
import { usePopoverArrow } from '../Popover/PopoverArrow.js';
import { u as useTooltipState } from '../TooltipState-17ceb977.js';

var useTooltipArrow = createHook({
  name: "TooltipArrow",
  compose: usePopoverArrow,
  useState: useTooltipState,
  useOptions: function useOptions(_ref) {
    var _ref$size = _ref.size,
        size = _ref$size === void 0 ? 16 : _ref$size,
        options = _objectWithoutPropertiesLoose(_ref, ["size"]);

    return _objectSpread2({
      size: size
    }, options);
  }
});
var TooltipArrow = createComponent({
  as: "div",
  memo: true,
  useHook: useTooltipArrow
});

export { TooltipArrow, useTooltipArrow };
