// reakit/lib/Composite/CompositeGroup.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var createComponent = require('reakit-system/createComponent');
var createHook = require('reakit-system/createHook');
require('../Box/Box.js');
var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-b18c4e5e.js');
var React = require('react');
var useForkRef = require('reakit-utils/useForkRef');
var useIsomorphicEffect = require('reakit-utils/useIsomorphicEffect');
require('reakit-utils/useSealedState');
require('reakit-utils/applyState');
require('../Id/IdProvider.js');
require('../Id/IdState.js');
var CompositeState = require('../CompositeState-969c2a43.js');
var Group = require('../Group/Group.js');
var Id = require('../Id/Id.js');

var unstable_useCompositeGroup = createHook.createHook({
  name: "CompositeGroup",
  compose: [Group.useGroup, Id.unstable_useId],
  useState: CompositeState.unstable_useCompositeState,
  useProps: function useProps(options, _ref) {
    var htmlRef = _ref.ref,
        htmlProps = _rollupPluginBabelHelpers._objectWithoutPropertiesLoose(_ref, ["ref"]);

    var ref = React.useRef(null);
    var id = options.id; // We need this to be called before CompositeItems' register

    useIsomorphicEffect.useIsomorphicEffect(function () {
      var _options$registerGrou;

      if (!id) return undefined;
      (_options$registerGrou = options.registerGroup) === null || _options$registerGrou === void 0 ? void 0 : _options$registerGrou.call(options, {
        id: id,
        ref: ref
      });
      return function () {
        var _options$unregisterGr;

        (_options$unregisterGr = options.unregisterGroup) === null || _options$unregisterGr === void 0 ? void 0 : _options$unregisterGr.call(options, id);
      };
    }, [id, options.registerGroup, options.unregisterGroup]);
    return _rollupPluginBabelHelpers._objectSpread2({
      ref: useForkRef.useForkRef(ref, htmlRef)
    }, htmlProps);
  }
});
var unstable_CompositeGroup = createComponent.createComponent({
  as: "div",
  useHook: unstable_useCompositeGroup
});

exports.unstable_CompositeGroup = unstable_CompositeGroup;
exports.unstable_useCompositeGroup = unstable_useCompositeGroup;
